src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -1087,17 +1098,17 @@ impl AuthProvider for OpenIDConnectAuthProvider {
     /// an error to report back to the user (one of the ApiAuth* Error types).
     /// Make sure to not leak any OIDC implementation details into the Error result!
     /// This function is also responsible for all logging around refreshing the token / extending the session.
-    fn authenticate(&self, request: &hyper::Request<hyper::Body>) -> KrillResult<Option<ActorDef>> {
+    pub async fn authenticate(&self, request: &hyper::Request<hyper::Body>) -> KrillResult<Option<ActorDef>> {
         trace!("Attempting to authenticate the request..");
 
-        self.initialize_connection_if_needed().map_err(|err| {
+        self.initialize_connection_if_needed().await.map_err(|err| {
             OpenIDConnectAuthProvider::internal_error(
                 "OpenID Connect: Cannot authenticate request: Failed to connect to provider",
                 Some(&stringify_cause_chain(err)),
             )
         })?;
 
-        let res = match self.get_bearer_token(request) {
+        let res = match AdminTokenAuthProvider::get_bearer_token(request) {
```

Comment:
Maybe ideally this would be pulled out as it's not specific to the AdminToken provider.

Reply:
moved to `commons::util::httpclient`.